python/tskit/trees.py

```diff
@@ -4549,6 +4549,62 @@ def subset(self, nodes, record_provenance=True):
         tables.subset(nodes, record_provenance)
         return tables.tree_sequence()
 
+    def union(
+        self,
+        other,
+        node_mapping,
+        check_shared_equality=True,
+        add_populations=True,
+        record_provenance=True,
+    ):
+        """
+        Returns an expanded tree sequence which contains the node-wise union of
+        ``self`` and ``other``. The operation adds the non-shared portions of
+        ``other`` onto a copy of ``self``. The indices in the ``node_mapping``
+        array correspond to node ids in ``other``, and the elements encode the
+        equivalent node id in ``self`` or ``tskit.NULL`` if the node is
+        exclusive to ``other``. Nodes that are exclusive ``other`` are added to
+        a copy of ``self``, along with:
+
+        1. Individuals which are new to ``self``.
+        2. Edges whose parent or child are new to ``self``.
+        3. Sites which were not present in ``self``.
```

Comment:
", if the site contains a newly added mutation." (or did you change this behavior?)

Reply:
that's right

Reply:
that's right: you changed it? Good idea, I suppose.

Reply:
yes! I've changed all your points except for the unresolved. some things I haven't pushed to github yet.